lib/prom/prom.go

```diff
@@ -0,0 +1,134 @@
+package prom
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+)
+
+//PrometheusMetrics vegeta metrics observer with exposition as Prometheus metrics endpoint
+type PrometheusMetrics struct {
+	requestSecondsHistogram *prometheus.HistogramVec
+	requestBytesInCounter   *prometheus.CounterVec
+	requestBytesOutCounter  *prometheus.CounterVec
+	requestFailCounter      *prometheus.CounterVec
+	srv                     http.Server
+	registry                *prometheus.Registry
+}
+
+//NewPrometheusMetrics same as NewPrometheusMetricsWithParams with default params:
+func NewPrometheusMetrics() (*PrometheusMetrics, error) {
+	return NewPrometheusMetricsWithParams("0.0.0.0:8880")
+}
+
+// NewPrometheusMetricsWithParams creates a new Prometheus Metrics to Observe attack results and expose metrics
+// For example, after using NewPrometheusMetricsWithParams("0.0.0.0:8880"),
+// during an "attack" you can call "curl http://127.0.0.0:8880" to see current metrics.
+// This endpoint can be configured in scrapper section of your Prometheus server.
+func NewPrometheusMetricsWithParams(bindURL string) (*PrometheusMetrics, error) {
+
+	//parse bind url elements
+	re := regexp.MustCompile("(.+):([0-9]+)")
+	rr := re.FindAllStringSubmatch(bindURL, 3)
```

Comment:
There are built-in functions for this. Use `url.Parse` and `net.SplitHostPort`

Reply:
ok. thanks for the advice!